pytest_plugins/logging_hooks.py

```diff
@@ -0,0 +1,69 @@
+import logging
+
+import logzero
+import pytest
+from xdist import is_xdist_worker
+
+from robottelo.logging import DEFAULT_DATE_FORMAT
+from robottelo.logging import logger
+from robottelo.logging import robottelo_log_dir
+from robottelo.logging import robottelo_log_file
+
+try:
+    from pytest_reportportal import RPLogger
+    from pytest_reportportal import RPLogHandler
+except ImportError:
+    pass
+
+
+@pytest.fixture(autouse=True, scope='session')
+def configure_logging(request, worker_id):
+    """Handle xdist and ReportPortal logging configuration at session start
+
+    Set up a separate logger for each pytest-xdist worker
+    if worker_id != 'master' then xdist is running in multi-threading so
+    a logfile named 'robottelo_gw{worker_id}.log' will be created.
+
+    Add a handler for ReportPortal logging
+    """
+    worker_formatter = logzero.LogFormatter(
+        fmt=f'%(asctime)s - {worker_id} - %(name)s - %(levelname)s - %(message)s',
+        datefmt=DEFAULT_DATE_FORMAT,
+    )
+    use_rp_logger = hasattr(request.node.config, 'py_test_service')
+    if use_rp_logger:
+        logging.setLoggerClass(RPLogger)
+
+    if is_xdist_worker(request):
```

Comment:
This restricts RP live logging to only run when using xdist. This is the same logic that was applied before - but it doesn't _have_ to be this way.  We generally will only be using RP result upload when using xdist in CI pipelines, which I think is why its setup this way. If you run without xdist and with RP options specified, it will upload results (in a tree, even if you set hierarchy to False), you just won't get any logs included.